j2k/testData/fileOrElement/issues/operationsWithChar.kt

```diff
@@ -42,6 +42,69 @@ internal class Test {
         i(c.toInt() ushr i)
     }
 
+    fun operationsWithCharLiteral() {
+        val i = 1
+
+        b(i > '0'.toInt())
+        b(i >= '0'.toInt())
+        b(i < '0'.toInt())
+        b(i <= '0'.toInt())
+
+        b('0'.toInt() > i)
+        b('0'.toInt() >= i)
+        b('0'.toInt() < i)
+        b('0'.toInt() <= i)
+
+        b('0'.toInt() == i)
+        b('0'.toInt() != i)
+
+        b(i == '0'.toInt())
+        b(i != '0'.toInt())
+
+        i(i + '0'.toInt())
+        i(i - '0'.toInt())
+        i(i / '0'.toInt())
+        i(i * '0'.toInt())
+        i(i % '0'.toInt())
+        i(i or '0'.toInt())
+        i(i and '0'.toInt())
+        i(i shl '0'.toInt())
+        i(i shr '0'.toInt())
+        i(i ushr '0'.toInt())
+
+        i('0'.toInt() + i)
+        i('0'.toInt() - i)
+        i('0'.toInt() / i)
+        i('0'.toInt() * i)
+        i('0'.toInt() % i)
+        i('0'.toInt() or i)
+        i('0'.toInt() and i)
+        i('0'.toInt() shl i)
+        i('0'.toInt() shr i)
+        i('0'.toInt() ushr i)
+    }
+
+    fun operationsWithCharLiterals() {
+        b('A'.toInt() > '0'.toInt())
+        b('A'.toInt() >= '0'.toInt())
+        b('A'.toInt() < '0'.toInt())
+        b('A'.toInt() <= '0'.toInt())
+
+        b('0'.toInt() == 'A'.toInt())
+        b('0'.toInt() != 'A'.toInt())
```

Comment:
This `toInt` calls are redundant. 
It should be possible to not generate it after 2be27d53800c56e2555d75420336904fe02161a8

Please, merge the latest master, and try to work around it.

Reply:
Sure thing @semoro, doing it now! :)
How do I know where the `toInt` is needed currently, as it seems the comparisons and the `minus` method don't need it, the rest does?

Reply:
Done